mesonbuild/interpreter/compiler.py

```diff
@@ -684,12 +685,24 @@ def has_multi_arguments_method(self, args: T.Sequence[str], kwargs: dict):
         return result
 
     @FeatureNew('compiler.get_supported_arguments', '0.43.0')
-    @permittedKwargs({})
-    def get_supported_arguments_method(self, args, kwargs):
+    @FeatureNewKwargs('compiler.get_supported_arguments', '0.59.0', ['checked'])
+    @typed_kwargs('compiler.get_supported_arguments', KwargInfo('checked', str, default='off'))
+    def get_supported_arguments_method(self, args: T.Sequence[str], kwargs: T.Dict[str, T.Any]):
         args = mesonlib.stringlistify(args)
         supported_args = []
+        checked = kwargs.pop('checked')
+
+        if checked not in ['warn', 'require', 'off']:
+            raise mesonlib.MesonException('"checked" kwarg must be one of "warn", "require" or "off"')
```

Comment:
`typed_kwargs` actually has you covered here:
```python
@typed_kwargs(
  'compiler.get_supported_args')
  KwargInro('checked', str, default='off', since='0.59.0',
             validator=lambda x: 'must be one of "warn", "require", or "off"' if x not in {'warn', 'require', 'off'} else None),
)
```
Thne you don't need `FeatureNewKwargs` or this check here, `typed_kwargs` will do that for you.

Reply:
Alright neat, I was not aware.